File: code/sorting/bogo_sort/bogosort.py
import random
 # Part of Cosmos by OpenGenus Foundation
def bogosort(l):
    while not in_order(l):
        random.shuffle(l)
    return l
 
def in_order(l):
    if not l:
        return True
    last = l[0]
    for x in l[1:]:
        if x < last:
            return False
        last = x
    return True


print(bogosort([5, 2, 3, 4, 1]))

File: code/sorting/bogo_sort/bogosort.fs
//Part of Cosmos by OpenGenus Foundation
open System
let rnd = Random()

//Simple method to shuffle lists, not really random in terms of crypthography
let shuffle (x:IComparable list)=
    List.sortBy(fun x -> rnd.Next()) x

let isSorted (sList:IComparable list)=
    let folder = fun (a,b) x -> (a && (b<=x),x)
    fst << List.fold (folder) (true,sList.[0]) <| sList

let rec bogosort (x: IComparable list)= 
    let shuffled = shuffle x
    if isSorted shuffled then
        shuffled
    else
        bogosort x

[<EntryPoint>]
let main (argv) =
    printfn "%A" <| bogosort [14;3;31;4;92;1]
    0
